grove/alpha/simon/simon.py

```diff
@@ -13,441 +13,334 @@
-    W, insert_row_num = make_square_row_echelon(W)
-
-    s = np.zeros(shape=(n,), dtype=int)
-    # inserted row is chosen not be orthogonal to s
-    s[insert_row_num] = 1
-
-    s = binary_back_substitute(W, s)
-
-    s_str = ''.join(str(x) for x in s)
-
-    return s_str, iterations, simon_program
-
-
-def check_two_to_one(cxn, oracle, ancillas, s):
-    """
-    Check if the oracle is one-to-one or two-to-one. The oracle is known
-    to represent either a one-to-one function, or a two-to-one function
-    with mask :math:`s`.
-
-    :param JobConnection cxn: the connection used to run programs
-    :param Program oracle: the oracle to query;
-                           emulates a classical :math:`f(x)`
-                           function as a blackbox.
-    :param list(int) ancillas: the ancillary qubits, where :math:`f(x)`
-                                is written to by the oracle
-    :param str s: the proposed mask of the function, found by Simon's algorithm
-    :return: true if and only if the oracle represents a function
-             that is two-to-one with mask :math:`s`
-    :rtype: bool
-    """
-    zero_program = oracle
-    mask_program = pq.Program()
-    for i in range(len(s)):
-        if s[i] == '1':
-            mask_program.inst(X(i))
-    mask_program += oracle
-
-    zero_value = cxn.run_and_measure(zero_program, ancillas)[0]
-    mask_value = cxn.run_and_measure(mask_program, ancillas)[0]
-
-    return zero_value == mask_value
-
-
-if __name__ == "__main__":
-    import pyquil.api as api
-
-    # Read function mappings from user
-    n = int(input("How many bits? "))
-    assert n > 0, "The number of bits must be positive."
-    print("Enter f(x) for the following n-bit inputs:")
-    mappings = []
-    for i in range(2 ** n):
-        val = input(np.binary_repr(i, n) + ': ')
-        assert all(list(map(lambda x: x in {'0', '1'}, val))), \
-            "f(x) must return only 0 and 1"
-        mappings.append(int(val, 2))
-
-    qvm = api.SyncConnection()
-
-    qubits = range(n)
-    ancillas = range(n, 2 * n)
-
-    unitary_funct = unitary_function(mappings)
-    oracle = oracle_function(unitary_funct, qubits, ancillas)
-
-    s, iterations, simon_program = find_mask(qvm, oracle, qubits)
-    two_to_one = check_two_to_one(qvm, oracle, ancillas, s)
-
-    if two_to_one:
-        print("The function is two-to-one with mask s = ", s)
-    else:
-        print("The function is one-to-one")
-    print("Iterations of the algorithm: ", iterations)
-
-    if input("Show Program? (y/n): ") == 'y':
-        print("----------Quantum Program Used----------")
-        print(simon_program)
+    def __init__(self):
+        self.unitary_function_mapping = None
+        self.n_qubits = None
+        self.n_ancillas = None
+        self._qubits = None
+        self.computational_qubits = None
+        self.ancillas = None
+        self.simon_circuit = None
+        self.oracle_circuit = None
```

Comment:
I don't know `oracle_circuit` is used anywhere.

Reply:
you're right!